lib/Service/LdapQuerent.php

```diff
@@ -109,10 +111,10 @@ public function find(string $search, int $limit = 0): \Generator {
 	}
 
 	/**
-	 * @throws \Exception
+	 * @throws Exception
 	 */
 	protected function getClient(): Ldap {
-		if ($this->ldap instanceof Ldap) {
+		if (isset($this->ldap) && $this->ldap instanceof Ldap) {
```

Comment:
Why this change? It is typed as `?Ldap`, the checks are redundant, the `instanceof` should be enough.

Reply:
Of course that's also possible, then it needs to be initialized as null though, otherwise there'll be a fatal error for it being uninitialized.

Reply:
It depends, I'd say. Not necessary when you set it in the constructor.

Reply:
Setting it in the constructor also counts as initializing, this is what I meant, once constructor is finished properties must respect their types.